examples/src/charmory_examples/object_detection/yolov5_license_plates.py

```diff
@@ -0,0 +1,210 @@
+import argparse
+from pprint import pprint
+
+import albumentations as A
+from art.attacks.evasion import RobustDPatch
+from art.estimators.object_detection import PyTorchYolo
+import datasets
+import numpy as np
+import torch
+from torchvision.ops import box_convert
+import yolov5
+from yolov5.utils.loss import ComputeLoss
+
+from armory.art_experimental.attacks.patch import AttackWrapper
+from armory.metrics.compute import BasicProfiler
+from charmory.data import ArmoryDataLoader
+from charmory.engine import EvaluationEngine
+import charmory.evaluation as ev
+from charmory.tasks.object_detection import ObjectDetectionTask
+from charmory.track import track_init_params, track_params
+
+
+def get_cli_args():
+    parser = argparse.ArgumentParser(
+        description="Run YOLOv5m object detection for license plates",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--batch-size",
+        default=4,
+        type=int,
+    )
+    parser.add_argument(
+        "--export-every-n-batches",
+        default=5,
+        type=int,
+    )
+    parser.add_argument(
+        "--num-batches",
+        default=20,
+        type=int,
+    )
+    return parser.parse_args()
+
+
+class Yolo(torch.nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.model = model
+        self.compute_loss = ComputeLoss(self.model.model.model)
+
+    def forward(self, x, targets=None):
+        if self.training:
+            outputs = self.model.model.model(x)
+            loss, _ = self.compute_loss(outputs, targets)
+            return dict(loss_total=loss)
+        return self.model(x)
+
+
+@track_params
+def load_model(
+    model_path="keremberke/yolov5m-license-plate",
+    conf=0.25,  # NMS confidence threshold
+    iou=0.45,  # NMS IoU threshold
+    agnostic=False,  # NMS class-agnostic
+    multi_label=False,  # NMS multiple labels per box
+    max_det=1000,  # maximum number of detections per image
+):
+    model = yolov5.load(model_path)
+    # set model parameters
+    model.conf = conf
+    model.iou = iou
+    model.agnostic = agnostic
+    model.multi_label = multi_label
+    model.max_det = max_det
+    return Yolo(model)
+
+
+def create_transform():
```

Comment:
Do you think we can ever take this function out of the examples? I guess it is slightly different depending on the dataset and maybe model being used. Can we create a generalized one for object detection or would it depend on the model too much?

Reply:
I've been thinking about this too. We probably can make some sort of abstraction/convenience utility, but it'll only ever be a 90% solution (as in, we can't anticipate all possible use-cases)--so I worry about making it part of the official armory-library API. I was talking to Matt the other day about the idea of separating armory-library core and another pip-installable package that provides these kinds of helpers & utilities. Though I guess there's not too much difference than putting them in a sub-package and just documenting them as "these might or might not help, but they're not required to be used either".

Reply:
Oh okay yea I think that would be useful b/c it can get confusing if you don't fully understand it